### PR TITLE
Unhide hidden members by renaming them and rebind Mesh enums

### PR DESCRIPTION
**Pull request by aaronfranke**
The entire purpose of this PR is to break compatibility. It benefits the scripting API due to not having members shadow one another, and it increases verbosity. For details, see https://github.com/godotengine/godot/issues/15763#issuecomment-568908661, but this PR does not close that issue since there are still a few other things.

* Rename XR `get_type` and `get_name` to `get_tracker_type` and `get_tracker_name`.

* Rename CPUParticles/CPUParticles2D/ParticlesMaterial `Flags` enum to `ParticleFlags`.

* Rename PathFollow2D `rotate` bool to `rotates`.

* Rebind Mesh/ArrayMesh enums based on where they are used and to not have a duplicate `ArrayType` enum.

* Rename LightOccluder2D `light_mask` to `occluder_light_mask` to avoid masking `light_mask`.

* Rename RD uniform `type` to `uniform_type` (newly exposed in April 2020).

* Rename RD texture `type` to `texture_type` (newly exposed sometime in mid-2020).

* Rename EditorNode3DGizmoPlugin `get_name` to `get_gizmo_name`.

In a nutshell, we should avoid ambiguous words like `name` and `type` and `flags`. For example, with RDUniform `type`, how do you know that "type" refers to a variable of type enum `RenderingDevice.UniformType`? Maybe it could be the type of the variable in Variant `TYPE` (so `Variant::INT`)? Maybe it could be some way to express the struct type `RDUniform`? What if you just have `var u` in GDScript and then `u.type` with no mention of "Uniform"? We should try to write things to be self-evident and not confusing. Also, `type` in a getter is `get_type`, which in C# would be `GetType()`, which conflicts with C#'s `GetType()`.

### CODE REVIEW

File: doc/classes/Mesh.xml [hunk 1]
```diff
@@ -184,31 +178,37 @@
...
 		</constant>
 		<constant name="ARRAY_INDEX" value="8" enum="ArrayType">
-			Array of indices.
+			[PackedInt32Array] of integers used as indices referencing vertices, colors, normals, tangents, and textures.
+			All of those arrays must have the same number of elements as the vertex array. No index can be beyond the
+			vertex array size. When this index array is present, it puts the function into "index mode," where the
+			index selects the *i*'th vertex, normal, tangent, color, UV, etc. This means if you want to have different
+			normals or colors along an edge, you have to duplicate the vertices.
+			For triangles, the index array is interpreted as triples, referring to the vertices of each triangle.
+			For lines, the index array is in pairs indicating the start and end of each line.
```
Comment (akien-mga) [inline doc/classes/Mesh.xml:211]: You should not using hard wrapping. Each line == one paragraph.

File: scene/resources/mesh.cpp [hunk 2]
```diff
@@ -480,9 +480,6 @@ void Mesh::_bind_methods() {
 	BIND_ENUM_CONSTANT(PRIMITIVE_TRIANGLES);
 	BIND_ENUM_CONSTANT(PRIMITIVE_TRIANGLE_STRIP);
 
-	BIND_ENUM_CONSTANT(BLEND_SHAPE_MODE_NORMALIZED);
-	BIND_ENUM_CONSTANT(BLEND_SHAPE_MODE_RELATIVE);
-
 	BIND_ENUM_CONSTANT(ARRAY_FORMAT_VERTEX);
```
Comment (akien-mga) [inline scene/resources/mesh.cpp:483]: All the other enum bindings here are also redundant (at least on current `master`).

File: scene/2d/cpu_particles_2d.cpp [hunk 3]
```diff
@@ -1279,8 +1279,8 @@ void CPUParticles2D::_bind_methods() {
 	ADD_PROPERTY(PropertyInfo(Variant::PACKED_VECTOR2_ARRAY, "emission_points"), "set_emission_points", "get_emission_points");
 	ADD_PROPERTY(PropertyInfo(Variant::PACKED_VECTOR2_ARRAY, "emission_normals"), "set_emission_normals", "get_emission_normals");
 	ADD_PROPERTY(PropertyInfo(Variant::PACKED_COLOR_ARRAY, "emission_colors"), "set_emission_colors", "get_emission_colors");
-	ADD_GROUP("Flags", "flag_");
-	ADD_PROPERTYI(PropertyInfo(Variant::BOOL, "flag_align_y"), "set_particle_flag", "get_particle_flag", FLAG_ALIGN_Y_TO_VELOCITY);
+	ADD_GROUP("ParticleFlags", "particle_flag_");
```
Comment (akien-mga) [inline scene/2d/cpu_particles_2d.cpp:1282]: Should be "Particle Flags". It's human readable group names.

File: scene/2d/light_occluder_2d.cpp [hunk 4]
```diff
@@ -237,12 +237,8 @@ Ref<OccluderPolygon2D> LightOccluder2D::get_occluder_polygon() const {
 }
 
 void LightOccluder2D::set_occluder_light_mask(int p_mask) {
-	mask = p_mask;
-	RS::get_singleton()->canvas_light_occluder_set_light_mask(occluder, mask);
-}
-
-int LightOccluder2D::get_occluder_light_mask() const {
-	return mask;
+	set_light_mask(p_mask);
```
Comment (akien-mga) [inline scene/2d/light_occluder_2d.cpp:240]: This is wrong IMO, you're now calling
```
        RS::get_singleton()->canvas_item_set_light_mask(canvas_item, p_light_mask);
```
which did not happen previously.

Those two properties have the same name but they're *not* the same, at least in the current API.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!
